File: samples/java/confluent/src/main/java/com/contoso/kafka/KafkaTriggerWithHeaders.java
package com.contoso.kafka;

import java.util.*;
import com.microsoft.azure.functions.annotation.*;
import com.contoso.kafka.entity.KafkaEntity;
import com.contoso.kafka.entity.KafkaHeaders;
import com.microsoft.azure.functions.*;

public class KafkaTriggerWithHeaders {
    @FunctionName("KafkaTriggerWithHeaders")
    public void runSingle(
            @KafkaTrigger(
                name = "KafkaTrigger",
                topic = "topic",  
                brokerList="%BrokerList%",
                consumerGroup="$Default", 
                username = "%ConfluentCloudUsername%", 
                password = "ConfluentCloudPassword",
                authenticationMode = BrokerAuthenticationMode.PLAIN,
                protocol = BrokerProtocol.SASLSSL,
                // sslCaLocation = "confluent_cloud_cacert.pem", // Enable this line for windows.
                dataType = "string"
             ) KafkaEntity kafkaEventData,
            final ExecutionContext context) {
            context.getLogger().info("Java Kafka trigger function called for message: " + kafkaEventData.Value);
            context.getLogger().info("Headers for the message:");
            for (KafkaHeaders header : kafkaEventData.Headers) {
                String decodedValue = new String(Base64.getDecoder().decode(header.Value));
                context.getLogger().info("Key:" + header.Key + " Value:" + decodedValue);                    
            }
    }
}

File: samples/dotnet-isolated/confluent/KafkaTriggerWithHeaders.cs
using System;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;
using Microsoft.Azure.Functions.Worker.Http;
using Newtonsoft.Json.Linq;


namespace Confluent
{
    public class KafkaTriggerWithHeaders
    {
        [Function("KafkaTriggerWithHeaders")]
        public static void Run(
            [KafkaTrigger("BrokerList",
                          "topic",
                          Username = "ConfluentCloudUserName",
                          Password = "ConfluentCloudPassword",
                          Protocol = BrokerProtocol.SaslSsl,
                          AuthenticationMode = BrokerAuthenticationMode.Plain,
                          ConsumerGroup = "$Default")] string eventData, FunctionContext context)
        {
            var eventJsonObject = JObject.Parse(eventData);
            var logger = context.GetLogger("KafkaFunction");
            logger.LogInformation($"C# Kafka trigger function processed a message: {eventJsonObject["Value"]}");
            logger.LogInformation($"Headers for this event: {eventJsonObject["Headers"]}");
        }
    }
}